ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts

```diff
@@ -0,0 +1,165 @@
+import { Page, test } from '@playwright/test';
+import {
+  BondTool,
+  BondType,
+  TopPanelButton,
+  clickOnAtom,
+  dragMouseTo,
+  drawBenzeneRing,
+  getControlModifier,
+  getCoordinatesOfTheMiddleOfTheScreen,
+  getCoordinatesTopAtomOfBenzeneRing,
+  openFileAndAddToCanvas,
+  selectNestedTool,
+  selectTopPanelButton,
+  takeEditorScreenshot,
+  waitForPageInit,
+} from '@utils';
+import { getAtomByIndex } from '@utils/canvas/atoms';
+import { getBondByIndex } from '@utils/canvas/bonds';
+
+test.describe('Rectangle selection tool', () => {
+  test.beforeEach(async ({ page }) => {
+    await waitForPageInit(page);
+  });
+
+  test.afterEach(async ({ page }) => {
+    await takeEditorScreenshot(page);
+  });
+
+  const xDelta = 30;
+  const yDelta = 60;
+  const modifier = getControlModifier();
+
+  const xMark = 300;
+  const yMark = 200;
+
+  async function selectObjects(page: Page, xAxis: number, yAxis: number) {
+    const point = await getCoordinatesOfTheMiddleOfTheScreen(page);
+    await page.mouse.move(point.x - xAxis, point.y - yAxis);
+    await page.mouse.down();
+    await page.mouse.move(point.x + xAxis, point.y + yAxis);
+    await page.mouse.up();
+    return point;
+  }
+
+  async function clickCanvas(page: Page) {
+    await page.mouse.click(xMark, yMark);
+  }
+
+  test('Structure selection with rectangle selection tool', async ({
+    page,
+  }) => {
+    // Test case: EPMLSOPKET-1347
+    await openFileAndAddToCanvas('KET/two-benzene-with-atoms.ket', page);
+    await page.getByTestId('select-rectangle').click();
+    await clickCanvas(page);
+    await selectObjects(page, xMark, yMark);
+  });
+
+  test('Drag structure', async ({ page }) => {
+    // Test case: EPMLSOPKET-1348
+    const objectSelection = 100;
+    const atomNumber = 5;
+    await openFileAndAddToCanvas('KET/two-benzene-with-atoms.ket', page);
+    await page.getByTestId('select-rectangle').click();
+    const point = await selectObjects(page, objectSelection, objectSelection);
+    await clickOnAtom(page, 'C', atomNumber);
+    await dragMouseTo(point.x + xDelta, point.y - yDelta, page);
+  });
+
+  test('Reaction components selection', async ({ page }) => {
+    //  Test case: EPMLSOPKET-1349
+    const atomNumber = 5;
+    const moveMouseCoordinatesY = 10;
+    const moveMouseCoordinatesX = 270;
+    await openFileAndAddToCanvas('Rxn-V2000/benzene-chain-reaction.rxn', page);
+    await page.getByTestId('select-rectangle').click();
+    const point = await getCoordinatesOfTheMiddleOfTheScreen(page);
+    await page.mouse.move(
+      point.x - moveMouseCoordinatesX,
+      point.y + moveMouseCoordinatesY,
+    );
+    await page.mouse.down();
```

Comment:
Why do we need mouse.down and then mouse.up? => it is a click
And then we again perform a click?

Reply:
done